plenum/server/client_authn.py

```diff
@@ -246,8 +246,11 @@ def authenticate(self, req_data, identifier: Optional[str]=None,
                           EmptyIdentifier):
                     ex = ex(req_data.get(f.IDENTIFIER.nm), req_data.get(f.SIG.nm))
                 raise ex
-        else:
+        if f.SIGS.nm in req_data:
             signatures = req_data[f.SIGS.nm]
+            identifier = identifier or req_data.get(f.IDENTIFIER.nm, None)
+            if identifier and identifier not in signatures:
+                raise IdentifierNotInSignatures(identifier, signatures)
```

Comment:
Do we need else if we don't have neither SIG nor SIGS? `signatures` will be uninitialized in this case, and it looks like we don't habe any other validation making sure that we must have either SIG or SIGS.